changelog/0007-reorganize-pillows.yml

```diff
@@ -0,0 +1,107 @@
+
+  Exact reorganization is as per below.
+
+  Old Pillow | New Pillow
+  -- | --
+  CaseToElasticsearchPillow | get_ucr_es_case_pillow
+  CaseSearchToElasticsearchPillow | get_ucr_es_case_pillow
+  kafka-ucr-static | get_ucr_es_case_pillow, get_ucr_es_form_pillow, get_user_es_ucr_pillow
+  XFormToElasticsearchPillow | get_ucr_es_form_pillow
+  FormSubmissionMetadataTrackerPillow | get_ucr_es_form_pillow
+  kafka-ucr-main | get_ucr_es_case_pillow, get_ucr_es_form_pillow, get_user_es_ucr_pillow
+  AppDbChangeFeedPillow | No Change
+  ApplicationToElasticsearchPillow | No Change
+  CacheInvalidatePillow | No Change
+  DefaultChangeFeedPillow | No Change
+  DomainDbKafkaPillow | No Change
+  FarmerRecordFluffPillow | No Change
+  GeographyFluffPillow | No Change
+  GroupPillow | get_group_es_pillow
+  GroupToUserPillow | get_group_es_pillow
+  KafkaDomainPillow | No Change
+  LedgerToElasticsearchPillow | No Change
+  M4ChangeFormFluffPillow | No Change
+  ReportCaseToElasticsearchPillow | get_ucr_es_case_pillow
+  ReportXFormToElasticsearchPillow | get_ucr_es_form_pillow
+  SqlSMSPillow | No Change
+  UCLAPatientFluffPillow | No Change
+  UnknownUsersPillow | get_ucr_es_form_pillow
+  UpdateUserSyncHistoryPillow | No Change
+  UserCacheInvalidatePillow | No Change
+  UserGroupsDbKafkaPillow | No Change
+  UserPillow | get_user_es_ucr_pillow  (includes user UCR processor)
+
+update_steps: |
+  Ensure that you have deployed a version of CommCare later than the version mentioned above
+  and that none of pillows for a given document type are too far behind from each other.
+
+  1. Replace old pillows with new pillows in pillow section of your app_processes.yml
+     Here's an example [pull request](https://github.com/dimagi/commcare-cloud/pull/2415)
+     Do not merge this until the switch is fully complete. Make sure you create enough
+     processes to handle your load.
+  2. Stop all pillows. You may have to be on above branch (for e.g. pillow-reorg)
+  ```bash   
+  cchq <env> fab stop_pillows --branch=pillow-reorg
+  ```
+  3. Run below django management command to create checkpoints for new pillows.
+  ```bash   
+  cchq <env> django-manage create_checkpoints_for_merged_pillows
+  ```
+  You can pass `--check` option to dry-run.
+  4. Deploy the change so that supervisor configuration is updated for the pillow switch.
+  ```bash   
+  cchq <env> ap deploy_commcarehq.yml --tags=services --limit=pillowtop --branch=pillow-reorg
+  ```
+  5. Run below command so that the supervisor process rereads the new config
+  ```bash   
+  cchq <env> update-supervisor-confs --branch=pillow-reorg
+  ```
+  6. Finally start all pillows.
+  ```bash    
+  cchq <env> fab start_pillows --branch=pillow-reorg
+  ```
+
+  You may do few checks to make sure the change is deployed succesfully.
+
+  1. Do a `sudo supervisorctl status all` on all pillow machines and make sure that new pillows are running succesfully and old pillows are not.
```

Comment:
could do `cchq <env> fab supervisorctl:"status all"` to run them all at once or `cchq <env> run-shell-command --limit=pillowtop supervisorctl status all -b` (I think)

Reply:
rather than `run-shell-command` you can do `cchq <env> service pillowtop status`

Reply:
Good idea, thanks.